packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1889,7 +1890,19 @@ export class MergeTree {
                 if (!nodesToUpdate.includes(pendingSegment.parent!)) {
                     nodesToUpdate.push(pendingSegment.parent!);
                 }
+                deltaSegments.push({
+                    segment:pendingSegment,
+                });
             });
+            if(this.mergeTreeMaintenanceCallback) {
```

Comment:
This is not related to your code, probably more a feedback to Tony:
Not big fan of IFs like that. I'd rather make mergeTreeMaintenanceCallback getter to always return something. If we do not have real callback, return empty callback. And I'd make it private and remove setter - it looks like all existing code using it can be changed to provide it in ctor of MergeTree.